src/debugger/Debugger.cpp

```diff
@@ -173,21 +224,73 @@ void Debugger::stopAtBreakpoint(ByteCodeBlock* byteCodeBlock, uint32_t offset, E
     ASSERT(m_activeObjects.size() == 0);
     m_stopState = ESCARGOT_DEBUGGER_IN_WAIT_MODE;
 
-    while (processIncomingMessages(state, byteCodeBlock))
+    while (processEvents(state, byteCodeBlock))
         ;
 
     m_activeObjects.clear();
     m_delay = ESCARGOT_DEBUGGER_MESSAGE_PROCESS_DELAY;
 }
 
-void Debugger::releaseFunction(const void* ptr)
+void DebuggerRemote::byteCodeReleaseNotification(const void* ptr)
 {
     // All messages which involves this pointer should be ignored until the confirmation arrives.
-    m_releasedFunctions.push_back(reinterpret_cast<uintptr_t>(ptr));
-    sendPointer(ESCARGOT_MESSAGE_RELEASE_FUNCTION, ptr);
+    if (enabled()) {
+        m_releasedFunctions.push_back(reinterpret_cast<uintptr_t>(ptr));
+        sendPointer(ESCARGOT_MESSAGE_RELEASE_FUNCTION, ptr);
+    }
+}
+
+void DebuggerRemote::exceptionCaught(String* message, SavedStackTraceDataVector& exceptionTrace)
+{
+    if (!enabled()) {
+        return;
+    }
+
+    sendType(ESCARGOT_MESSAGE_EXCEPTION);
+
+    if (!enabled()) {
+        return;
+    }
+
+    sendString(ESCARGOT_MESSAGE_STRING_8BIT, message);
+
+    size_t size = exceptionTrace.size();
+    for (size_t i = 0; i < size && enabled(); i++) {
+        sendBacktraceInfo(ESCARGOT_MESSAGE_EXCEPTION_BACKTRACE, exceptionTrace[i].byteCodeBlock, exceptionTrace[i].line, exceptionTrace[i].column, UINT32_MAX);
```

Comment:
It looks like `depth` information for `ESCARGOT_MESSAGE_EXCEPTION_BACKTRACE` is set to `UINT32_MAX` value here and there is no update on it.